Refactor cloud prepare flags docs

dfarrell07 opened the pull request:
See commit messages for details.

Closes #848

dfarrell07 committed:
Update `cloud prepare` global flags to current

Update the documented global cloud prepare flags to all the ones
currently printed by `subctl cloud prepare --help`.

Relates-to: #848
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Move global cloud prepare flags to top of section

As the global flags apply to all cloud prepare subcommands, it makes
sense to show them to users at the top of the cloud parpare section,
before they find the specific gcp|aws... target docs and stop reading.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Move vxlan/nat ports to global cloud prep section

These flags are not shown in the Global Flags section of `subctl cloud
prepare --help`, but they are shown in the Flags section of that command
and all subcommands, so they are effectively global/shared.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Refactor `cloud prepare rhos` flags

There are many differences between the subctl output and the docs.
Try to take the best from both, and update both to be the same.

Pairs with a PR to subctl to update the command's docs.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Refactor cloud prepare aws|gcp|rhos|generic docs

Tweaks to the subctl cloud prepare subcommand flag docs to match the
subctl outputs.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Change cloud prep flag docs IPSec->IPsec

Unfortunately our code linting enforces IPSec as an initialism while our
MD linting enforces IPsec (which is what the project uses).

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

skitt reviewed (changes requested):
My goal with https://github.com/submariner-io/submariner-website/pull/845 was to document the new connection options in a single location, so that we wouldn’t need to repeat them in every section. This *doesn’t* document the various `--as` etc. options in detail, but the reference for those is `kubectl`.

That way, we can concentrate on only the relevant options in each `subctl` command, instead of drowning them in the `kubectl`-provided options.

dfarrell07 committed:
Remove connection flags from cloud prep flags

The connection flags are shared among many commands and should be
documented in a single place, not specific to cloud prepare. Per
feedback during code review, the preference seems to be to not
explicitly document them here but to use subctl as a reference.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
> My goal with #845 was to document the new connection options in a single location, so that we wouldn’t need to repeat them in every section. This _doesn’t_ document the various `--as` etc. options in detail, but the reference for those is `kubectl`.
> 
> That way, we can concentrate on only the relevant options in each `subctl` command, instead of drowning them in the `kubectl`-provided options.

I added e1bffa6ecb3743970b45b1dd94fc5cccc1464d41 to address this, let me know if that's what you had in mind.

skitt commented on src/content/operations/deployment/subctl/_index.en.md:
```diff
@@ -397,26 +397,9 @@ This command prepares the underlying cloud infrastructure for Submariner install
-| `--client-key` `<string>`               | Path to a client key file for TLS
-| `--cluster` `<string>`                  | The name of the kubeconfig cluster to use
-| `--context` `<string>`                  | The name of the kubeconfig context to use
 | `--insecure-skip-tls-verify` `<string>` | If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
```
This is a connection flag too.

dfarrell07 replied:
@sridhargaddam's original issue (https://github.com/submariner-io/submariner-website/issues/848) requested in particular `--insecure-skip-tls-verify` as a flag that should be added, so CCing him.

sridhargaddam replied:
Right, the same flag is supported even for `subctl join ...` command as well. I'm fine if you want to include it in this PR or as a separate commit/PR. Thanks for looking into it @dfarrell07

skitt reviewed (changes requested):
Thanks, this is what I had in mind; there’s just one more flag to remove.

nyechiel commented:
@dfarrell07 are you looking into this?

dfarrell07 commented:
> @dfarrell07 are you looking into this?

@nyechiel I'm not sure what to do. As I understand the feedback, @skitt is saying the flag should not be documented here (instead, reference `subctl` output) and @sridhargaddam's original issue requests for it to be explicitly documented.

dfarrell07 commented:
> > @dfarrell07 are you looking into this?
> 
> @nyechiel I'm not sure what to do. As I understand the feedback, @skitt is saying the flag should not be documented here (instead, reference `subctl` output) and @sridhargaddam's original issue requests for it to be explicitly documented.

For now, I removed the common flag here and documented that decision on the original request issue (https://github.com/submariner-io/submariner-website/issues/848#issuecomment-1412487154).
